app/src/main/java/com/epfl/drawyourpath/userProfileCreation/SelectUsernameFragment.kt

```diff
@@ -0,0 +1,163 @@
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.Button
+import android.widget.EditText
+import android.widget.TextView
+import androidx.fragment.app.Fragment
+import com.epfl.drawyourpath.R
+import com.epfl.drawyourpath.authentication.Auth
+import com.epfl.drawyourpath.authentication.MockAuth
+import com.epfl.drawyourpath.database.Database
+import com.epfl.drawyourpath.database.FirebaseDatabase
+import com.epfl.drawyourpath.database.MockDatabase
+import java.util.concurrent.CompletableFuture
+
+class SelectUsernameFragment : Fragment(R.layout.fragment_select_username) {
+
+    private var isTest: Boolean = false
+
+    private lateinit var database: Database
+    private lateinit var auth: Auth
+
+    private lateinit var outputView: TextView
+    private lateinit var inputUserName: EditText
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // retrieve the value from the welcome activity to know if we are running testes
+        isTest = arguments?.getBoolean(PROFILE_TEST_KEY) ?: false
+
+        setupEnv(isTest)
+
+        // select the correct database in function of test scenario
+        val testUserNameButton: Button =
+            view.findViewById(R.id.testUserName_button_userProfileCreation)
+        inputUserName = view.findViewById(R.id.input_userName_text_UserProfileCreation)
+
+        outputView = view.findViewById(R.id.testUserName_text_userProfileCreation)
+
+        testUserNameButton.setOnClickListener { checkAvailability(inputUserName.text.toString()) }
+
+        val setUserNameButton =
+            view.findViewById<Button>(R.id.setUserName_button_userProfileCreation)
+        setUserNameButton.setOnClickListener { onSetUsernameButtonClicked() }
+    }
+
+    private fun onSetUsernameButtonClicked() {
+        // try to set the userName to the database
+        val usernameStr = inputUserName.text.toString()
+
+        Log.i("DYP", "Checking availability of username $usernameStr")
+
+        checkAvailability(usernameStr).thenAccept {
+            if (it) {
+                showPersonalInfoFragment(usernameStr)
+                Log.i("DYP", "Username $usernameStr is available")
+            } else {
+                Log.e("DYP", "Username $usernameStr is taken")
+            }
+        }.exceptionally {
+            Log.e("DYP", "Failed to check username: ${it.message}")
+            it.printStackTrace()
+            setErrorMessage("${it.message}")
+            null
+        }
+    }
+
+    private fun setupEnv(isTest: Boolean) {
+        when (isTest) {
+            true -> {
+                Log.i("DYP", "SelectUsername fragment uses mock API.")
+
+                database = MockDatabase()
+                auth = MockAuth(forceSigned = true)
+            }
+            false -> {
+                database = FirebaseDatabase()
+                auth = MockAuth()
+            }
```

Comment:
The authentification Gove here is not the correct one. It should be the FireAuthentification .

Reply:
I have tested to create an account, and it was blocking the profile creation.